azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -331,78 +338,220 @@ def __init__(self, xml_text):
+                handler_name, version)
+            raise ExtensionConfigError(msg)
+
+        if ExtensionsConfig.__is_not_empty(runtime_settings_nodes):
+            if len(runtime_settings_nodes) > 1:
+                msg = "Multiple RuntimeSettings found for the same handler: {0} and version: {1} (Expected: 1; Available: {2})".format(
+                    handler_name, version, len(runtime_settings_nodes))
+                raise ExtensionConfigError(msg)
+            # Only Runtime settings available, parse that
+            ExtensionsConfig.__parse_runtime_settings(plugin_settings_node, runtime_settings_nodes[0], handler_name,
+                                                      ext_handler)
+        elif ExtensionsConfig.__is_not_empty(extension_runtime_settings_nodes):
+            # Parse the ExtensionRuntime settings for the given extension
+            ExtensionsConfig.__parse_extension_runtime_settings(plugin_settings_node, extension_runtime_settings_nodes,
+                                                                ext_handler)
 
+    @staticmethod
+    def __get_dependency_level_from_node(depends_on_node, name):
         depends_on_level = 0
-        depends_on_node = find(settings[0], "DependsOn")
-        if depends_on_node != None:  # pylint: disable=C0121
+        if depends_on_node is not None:
             try:
                 depends_on_level = int(getattrib(depends_on_node, "dependencyLevel"))
             except (ValueError, TypeError):
                 logger.warn("Could not parse dependencyLevel for handler {0}. Setting it to 0".format(name))
                 depends_on_level = 0
+        return depends_on_level
+
+    @staticmethod
+    def __parse_runtime_settings(plugin_settings_node, runtime_settings_node, handler_name, ext_handler):
+        """
+        Sample Plugin in PluginSettings containing DependsOn and RuntimeSettings (single settings per extension) -
+
+        <Plugin name="Microsoft.Compute.VMAccessAgent" version="2.4.7">
+        <DependsOn dependencyLevel="2">
+          <DependsOnExtension extension="firstRunCommand" handler="Microsoft.CPlat.Core.RunCommandHandlerWindows" />
+          <DependsOnExtension handler="Microsoft.Compute.CustomScriptExtension" />
+        </DependsOn>
+        <RuntimeSettings seqNo="1">{
+              "runtimeSettings": [
+                {
+                  "handlerSettings": {
+                    "protectedSettingsCertThumbprint": "<Redacted>",
+                    "protectedSettings": "<Redacted>",
+                    "publicSettings": {"UserName":"test1234"}
+                  }
+                }
+              ]
+            }
+        </RuntimeSettings>
+        </Plugin>
+        """
+        depends_on_nodes = findall(plugin_settings_node, "DependsOn")
+        if len(depends_on_nodes) > 1:
+            msg = "Extension Handler can only have a single dependsOn node for Single config extensions. Found: {0}".format(
```

Comment:
is this accurate? (we can sync offline)

Reply:
That is my understanding yes. I can confirm from CRP code

Reply:
never mind - This is DependsOn, not DependsOnExtension